snapcraft/internal/parts.py

```diff
@@ -273,7 +273,13 @@ def load_plugin(self, part_name, plugin_name, part_properties):
             part_schema=self._validator.part_schema,
             definitions_schema=self._validator.definitions_schema)
 
-        self.build_tools += part.code.build_packages
+        # Add multi-arch suffix for cross-compilation to dev packages
+        for package in part.code.build_packages:
+            if self._project_options.is_cross_compiling:
+                if package.endswith('-dev'):
```

Comment:
I'm pretty sure that the whole approach of appending architecture suffixes is wrong here, not least because apt has a whole load of non-trivial metadata about which packages need which kind of multi-arch treatment that's being ignored.  Just checking for `-dev` suffixes is going to go wrong in both directions.

Here's the [relevant spec](https://wiki.ubuntu.com/MultiarchCross) for the system-level behaviour.  To use this correctly, you need to somehow arrange to call `apt-get build-dep -a<target-arch>` (the behaviour of `apt-get build-dep` behaves differently in subtle but important ways from `apt-get install` - for instance the `:native` suffix isn't defined for ordinary dependencies, only build-dependencies).

My recommendation would be a somewhat different strategy, which is a bit more work but should be a lot more robust: generate a temporary `.dsc` file with the contents of `build-packages` in its `Build-Depends`, and pass that to `apt-get build-dep` or `apt-get build-dep -a<target-arch>` depending on whether you're cross-compiling.  On xenial you can just pass the `.dsc` directly to `apt-get build-dep`, but if you need to support trusty you need to faff about with turning it into a `Sources` file the same kind of way that `sbuild` does.

If it were me then I'd do that in a separate PR and rebase this one on top of it later ...